src/controller/ReadInteraction.h

```diff
@@ -218,11 +224,13 @@ CHIP_ERROR ReportEvent(Messaging::ExchangeManager * apExchangeMgr, EndpointId en
 template <typename DecodableEventType>
 CHIP_ERROR ReadEvent(Messaging::ExchangeManager * exchangeMgr, const SessionHandle & sessionHandle, EndpointId endpointId,
                      typename TypedReadEventCallback<DecodableEventType>::OnSuccessCallbackType onSuccessCb,
-                     typename TypedReadEventCallback<DecodableEventType>::OnErrorCallbackType onErrorCb)
+                     typename TypedReadEventCallback<DecodableEventType>::OnErrorCallbackType onErrorCb,
+                     typename TypedReadEventCallback<DecodableEventType>::OnSubscriptionEstablishedCallbackType onDoneCb = nullptr)
```

Comment:
If the expectation of this function is that you're trying to read out a specific type of event, and if you don't actually get any of those events, shouldn't that be interpreted as an error (in the context of this function)?

If so, I don't think adding a onDone cb is quite right - rather, it should call `onErrorCb`.